### PR TITLE
Make search admin controllers use the search service client

### PR DESCRIPTION
**Pull request by gmethvin**
This is just a rough split that allows the shoebox server to get the data from the search server through an the internal controller API.

### CODE REVIEW

File: shoebox/app/com/keepit/controllers/admin/AdminArticleIndexerController.scala [hunk 1]
```diff
@@ -6,22 +6,22 @@ import com.keepit.common.db._
...
   ) extends AdminController {
 
   def index = AdminHtmlAction { implicit request =>
-    val cnt = indexerPlugin.index()
-    Ok("indexed %d articles".format(cnt))
+    Async {
+      searchClient.index().map { cnt =>
+        Ok("indexed %d articles".format(cnt))
+      }
+    }
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/admin/AdminArticleIndexerController.scala:24]: Should we make async the default mode of our service clients? They could always return a `Future`.

I don't know whether most calls will be async or not (I suspect so), but I'm pretty sure Scala/Play make it easier to go from async → sync than the other way around.

File: shoebox/app/com/keepit/controllers/admin/AdminArticleIndexerController.scala [hunk 2]
```diff
@@ -29,31 +29,30 @@ class ArticleIndexerController @Inject()(
...
 
   def indexInfo = AdminHtmlAction { implicit request =>
-    Ok(html.admin.indexer(indexer))
+    Ok(searchClient.articleIndexInfo())
   }
 
   def getSequenceNumber = AdminJsonAction { implicit request =>
-    Ok(JsObject(Seq("sequenceNumber" -> JsNumber(indexer.sequenceNumber.value))))
+    val number = searchClient.articleIndexerSequenceNumber()
+    Ok(JsObject(Seq("sequenceNumber" -> JsNumber(number))))
```
Comment (2is10) [inline shoebox/app/com/keepit/controllers/admin/AdminArticleIndexerController.scala]: May as well convert to the more concise pattern while we're here:
`Ok(Json.obj("sequenceNumber" -> number))`

**Comment by 2is10** [comment]
I'm happy with this. Great work! Feel free to merge when it's ready.

As we discussed in person, it might be fun to write an Action class that accepts a `Result` or a `Future[Result]`. Or an implicit converter from a `Future[Result]` to an `AsyncResult` or something.

**Comment by gmethvin** [comment]
I'm going to change all the calls to return futures and then merge

**Pull request merged**